.github/workflows/ci.yml

```diff
@@ -0,0 +1,178 @@
+
+################################## BUILD AND DOCS LAYER ########################
+
+  build:
+    needs: [rustfmt-clippy, prepare-checks]
+
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+        rust: [stable, prev]
+
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Debug matrix and previous
+        run: |
+          echo "Matrix rust: ${{ matrix.rust }}"
+          echo "Prepare-checks rust-prev-version: ${{ needs.prepare-checks.outputs.rust-prev-version }}"
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: ${{ matrix.rust == 'prev' && needs.prepare-checks.outputs.rust-prev-version || 'stable' }}
+
+      - name: Build
+        run: cargo build --verbose
+
+  docs:
+    needs: [build]
+
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Generate docs
+        run: cargo doc --verbose --no-deps
+
+################################## CODE COVERAGE LAYER ##################################
+
+  code-coverage:
+    needs: [docs]
+
+    runs-on: ubuntu-latest
+
+    env:
+      RUSTFLAGS: "-Cinstrument-coverage"
+      LLVM_PROFILE_FILE: "ascot-%p-%m.profraw"
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Install grcov
+        env:
+          GRCOV_BASE_URL: https://github.com/mozilla/grcov/releases
+          GRCOV_BINARY: grcov-x86_64-unknown-linux-musl.tar.bz2
+        run: |
+          # Get the latest release tag by:
+          # 1. Obtaining the URL of the latest release,
+          # 2. Extracting the semantic version from it,
+          # 3. Taking the first match found.
+          LATEST_TAG=$(curl -sIL -o /dev/null -w '%{url_effective}' "$GRCOV_BASE_URL/latest" | grep -oE 'v?[0-9]+\.[0-9]+\.[0-9]+' | head -n1)
+          GRCOV_URL="$GRCOV_BASE_URL/download/$LATEST_TAG/$GRCOV_BINARY"
+          curl -L "$GRCOV_URL" | tar xj -C $HOME/.cargo/bin
+
+      - name: Install llvm-tools-preview
+        run: rustup component add llvm-tools-preview
+
+      - name: Run ascot tests
+        run: |
+          set -euxo pipefail
+          cargo test --verbose -p ascot
+          cargo test --verbose -p ascot --no-default-features
+          cargo test --verbose -p ascot --no-default-features --features "alloc"
+
+      - name: Run ascot-os tests
+        run: |
+          set -euxo pipefail
+          cargo test --verbose -p ascot-os
+          cargo test --verbose -p ascot-os --no-default-features --features "mdns-sd-service"
```

Comment:
For now we can remove this command because we have just one feature and `--no-default-features` is not allowed for `ascot-os`